### ISSUE TITLE
[MSHARED-32] Customization of jar-name pattern of classpath-entries inside MANIFEST.MF

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-archiver#224]
**[manuel aldana](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=aldana)** opened **[MSHARED-32](https://issues.apache.org/jira/browse/MSHARED-32?redirect=false)** and commented

i followed manifest-customation as explained in:
http://maven.apache.org/shared/maven-archiver/index.html

what i really miss, is to configure the classpath pattern of the dependency jar-names. the only customization of classpath entries is possible by setting the classpath prefix (\<addClasspath>) and (\<classpathMavenRepositoryLayout/>). unfortunately it is not possible to set the jar-name pattern itself.

to avoid library name-clashes we got pattern $groupId-$artifactId-$version for jars which are put besides the main-jar. by default maven-archiver only sets $artifactId-$version. so it would be nice to have a possible configuration like \<jarPattern> or similar inside maven archive-configuration .

---

**Issue Links:**
- [MJAR-28](https://issues.apache.org/jira/browse/MJAR-28) Using the jar plugin with addClasspath and snapshots can create manifest classpath with incorrect jar versions

- [MASSEMBLY-211](https://issues.apache.org/jira/browse/MASSEMBLY-211) assembly plugin and jar plugin disagree about whether to use uniqueVersion snapshot names
 (_**"is depended upon by"**_)

**Comment by jira-importer** [comment]
**[John Dennis Casey](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jdcasey)** commented

Deprecated \<classpathMavenRepositoryLayout> in favor of new \<classpathLayoutType>, which uses a value of 'repository' to mimick the \<classpathMavenRepositoryLayout>true\</classpathMavenRepositoryLayout> setting. For a custom layout, try:

```xml
<classpathLayoutType>custom</classpathLayoutType>
<customClasspathLayout>${artifact.artifactId}-${artifact.baseVersion}${dashClassifier?}.${artifact.extension}</customClasspathLayout>
```

The configuration for this should show up in the classpath examples page on the maven-archiver website once the release is done.
